resources/js/pages/CoursePlanningPage/components/PersonTable/PersonTableCell.vue

```diff
@@ -0,0 +1,192 @@
+          @click="isShowingAddCourse = true"
+        >
+          + Add Course
+        </button>
+      </template>
+    </Draggable>
+
+    <EditDraftSectionModal
+      v-if="isShowingAddCourse"
+      :initialPerson="person"
+      :initialTerm="term"
+      :initialRole="initialRole"
+      :show="isShowingAddCourse"
+      @close="isShowingAddCourse = false"
+      @save="handleSaveTentativeCourse"
+    />
+  </div>
+</template>
+<script setup lang="ts">
+import LeaveChip from "../LeaveChip.vue";
+import SectionDetails from "./SectionDetails.vue";
+import { ref, computed } from "vue";
+import EditDraftSectionModal from "../EditDraftSectionModal.vue";
+import * as T from "@/types";
+import { useRootCoursePlanningStore } from "../../stores/useRootCoursePlanningStore";
+import Draggable, { type DraggableChangeEvent } from "vuedraggable";
+import {
+  getElementFromDraggableEvent,
+  isDraggableAddedEvent,
+  isDraggableRemovedEvent,
+} from "@/utils/draggableHelpers";
+import { watchDebounced } from "@vueuse/core";
+import { $can } from "@/utils";
+
+const props = defineProps<{
+  person: T.Person;
+  term: T.Term;
+}>();
+
+const coursePlanningStore = useRootCoursePlanningStore();
+
+const courseSections = computed(() => {
+  return coursePlanningStore.getSectionsForEmplIdInTerm(
+    props.person.emplid,
+    props.term.id,
+  );
+});
+
+// use local course sections to avoid section jumping back
+// to original position while api call is made
+const localCourseSections = ref<T.CourseSection[]>(courseSections.value);
+
+watchDebounced(
+  courseSections,
+  (newCourseSections) => {
+    localCourseSections.value = newCourseSections;
+  },
+  { debounce: 300, maxWait: 1000 },
+);
+
+const isShowingAddCourse = ref(false);
+
+const termLeavesForPerson = computed(() =>
+  coursePlanningStore.leaveStore
+    .getLeavesByPersonId(props.person.id)
+    .filter((leave) => leave.termIds?.includes(props.term.id)),
+);
+
+const doesTermHavePublishedSections = computed(() => {
+  return (
+    coursePlanningStore.courseSectionStore
+      .getSectionsByTermId(props.term.id)
+      .filter((section) => section.isPublished).length > 0
+  );
+});
+
+const isPlannable = computed(() => {
+  return (
+    coursePlanningStore.isInPlanningMode && !doesTermHavePublishedSections.value
+  );
+});
+
+const isEditable = computed(() => {
+  return (
+    coursePlanningStore.isInPlanningMode &&
+    $can("edit planned courses") &&
+    !doesTermHavePublishedSections.value
+  );
+});
+
+const initialRole = computed(() => {
+  // if unset, check the filters to see if we're filtering by a single role
+  const filterRoles = coursePlanningStore.filters.includedEnrollmentRoles;
+  if (filterRoles.length === 1) {
+    return filterRoles[0];
+  }
+
+  // otherwise, default to Primary Instructor
+  return "PI";
+});
+
+function handleSaveTentativeCourse({ term, course, person, role }) {
+  console.log("handleSaveTentativeCourse");
```

Comment:
there are a few console.logs in this PR - leaving those in?

Reply:
it's an aesthetic choice.  😄 

Actually, there is one console.log that I'd like to leave in (not this one). I'm logging some performance metrics on report page: the time from api call completion to table mounting, which should give a good estimate of the time it takes to compute. Things like changing from v-show to v-if, or precomputing a lookup table for a getter seem to have a significant impact on performance.